### PR TITLE
Lucas/refactor directory structure

### PR DESCRIPTION
**Pull request by lgrossi**

### CODE REVIEW

**Review by dudantas — APPROVED** [review]
Tested, everything ok!

File: src/creatures/npcs/npc.cpp [hunk 1]
```diff
@@ -96,29 +97,15 @@ void Npc::onCreatureAppear(Creature* creature, bool isLogin)
...
-		lua_State* L = scriptInterface->getLuaState();
-		scriptInterface->pushFunction(npcType->info.creatureAppearEvent);
-
-		LuaScriptInterface::pushUserdata<Npc>(L, this);
-		LuaScriptInterface::setMetatable(L, -1, "Npc");
-
-		LuaScriptInterface::pushUserdata<Creature>(L, creature);
-		LuaScriptInterface::setCreatureMetatable(L, -1, creature);
+	// onCreatureAppear(self, creature)
+	CreatureCallback* callback = new CreatureCallback(npcType->info.scriptInterface, this);
```
Comment (Costallat) [inline src/creatures/npcs/npc.cpp]: check if it is possible to use a local variable

File: src/creatures/players/player.cpp [hunk 2]
```diff
@@ -17,24 +17,24 @@
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
-#include "otpch.h"
+#include "../../otpch.h"
```
Comment (Costallat) [inline src/creatures/players/player.cpp]: player.cpp needs to include player.h

File: src/lua/callbacks/creaturecallback.cpp [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+		LuaScriptInterface::pushUserdata<Npc>(L, npc);
+	}else if (Monster* monster = creature->getMonster()) {
+		LuaScriptInterface::pushUserdata<Monster>(L, monster);
+	}else if (Player* player = creature->getPlayer()) {
+		LuaScriptInterface::pushUserdata<Player>(L, player);
+	} else {
+		return;
+	}
+
+	params++;
```
Comment (Costallat) [inline src/lua/callbacks/creaturecallback.cpp:59]: check here

**Review by Costallat — APPROVED** [review]

**Review by dex-89 — APPROVED** [review]

**Pull request merged**